cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -1918,7 +1921,11 @@ class AzureFileSystem::Impl {
           if (path.Name == base_location.path && !path.IsDirectory) {
             return NotADir(base_location);
           }
-          acc_results->push_back(FileInfoFromPath(base_location.container, path));
+          if (internal::GetAbstractPathDepth(path.Name) -
+                  internal::GetAbstractPathDepth(base_location.path) - 1 <=
```

Comment:
Can we compute `internal::GetAbstractPathDepth(base_location.path)` out of this loop and use the computed result here?

And can we simplify this something like this?

```cpp
auto max_depth = select.max_recursion + internal::GetAbstractPathDepth(base_location.path) + 1;
for () {
  if (internal::GetAbstractPathDepth(path.Name) <= max_depth) {
  }
}
```

BTW, why do we need `-1` here? Could you add a comment about what `-1` is needed for?

Reply:
Yeah, that's more readable :+1: . I've also added a comment about the off by one correction.

Reply:
Wait... actually that doesn't work because `select.max_recursion` defaults to `INT32_MAX` so this causes an integer overflow